src/main/java/com/zendesk/maxwell/schema/SchemaScavenger.java

```diff
@@ -0,0 +1,83 @@
+package com.zendesk.maxwell.schema;
+
+import com.zendesk.maxwell.RunLoopProcess;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import snaq.db.ConnectionPool;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SchemaScavenger extends RunLoopProcess implements Runnable {
+	static final Logger LOGGER = LoggerFactory.getLogger(SchemaStore.class);
+	private static final long MAX_ROWS_PER_SECOND = 500;
+	private final ConnectionPool connectionPool;
+
+	public SchemaScavenger(ConnectionPool pool) {
+		this.connectionPool = pool;
+	}
+
+	private List<Long> getDeletedSchemas() throws SQLException {
+		ArrayList<Long> list = new ArrayList<>();
+		try ( Connection connection = connectionPool.getConnection() ) {
+			ResultSet rs = connection.createStatement().executeQuery("select id from `maxwell`.`schemas` where deleted = 1 LIMIT 100");
+
+			while ( rs.next() ) {
+				list.add(rs.getLong("id"));
+			}
+		}
+
+		return list;
+	}
+
+	public void deleteSchema(Long id, Long maxRowsPerSecond) throws SQLException {
+		String[] tables = { "columns", "tables", "databases" };
+
+		try ( Connection connection = connectionPool.getConnection() ) {
+			connection.createStatement().execute("delete from `maxwell`.`schemas` where id = " + id);
```

Comment:
i am a bit worried about deleting the id before we get all the rows. Can't this be done after?